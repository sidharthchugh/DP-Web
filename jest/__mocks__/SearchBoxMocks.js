import React from 'react';
import FontAwesome from 'react-fontawesome';
import {Creatable} from 'react-select';
import DropdownList from 'react-widgets/lib/DropdownList';
import Geosuggest from 'react-geosuggest';
import DateTimePicker from 'react-widgets/lib/DateTimePicker';
import NumberPicker from 'react-widgets/lib/NumberPicker';
import Textarea from 'react-textarea-autosize';

const options = [
    { value: 'one', label: 'One' },
    { value: 'two', label: 'Two' }
];

const mockChooseEditType = (value, i) => {
  return (
    <div className="edit-field">
      {(() => {
        if (value.type === 'dropdown') {
          return (
            <DropdownList name={value.name} data={value.typeValues} placeholder={value.label} />
          );
        } else if (value.type === 'dropdownSearch') {
          return (
            <DropdownList name={value.name} data={value.typeValues} caseSensitive={false} minLength={2} filter="contains" placeholder={value.label} />
          );
        } else if (value.type === 'string') {
          return (
            <div>
              <label>{value.label}</label>
              <div>
                <input placeholder={value.label} type={value.type} />
              </div>
            </div>
          );
        } else if (value.type === 'number') {
          return (
            <NumberPicker name={value.name} placeholder={value.label} />
          );
        } else if (value.type === 'calandar') {
          return (
            <DateTimePicker name={value.name} placeholder={value.label} />
          );
        } else if (value.type === 'phoneNumber') {
          return (
            <div>
              <label>{value.label}</label>
              <div>
                <input placeholder={value.label} type={'number'} />
              </div>
            </div>
          );
        } else if (value.type === 'multiSelect') {
         return (
           <Creatable multi value={null} options={options} onChange={() => {}} placeholder={value.label} />
          );
        } else if (value.type === 'textarea') {
          return (
            <Textarea placeholder={value.label} defaultValue={''} onChange={() => {}} minRows={3} maxRows={6} />
          );
        } else if (value.type === '(cities)' || value.type === '(regions)' || value.type === 'address') {
          return (
            <Geosuggest placeholder={value.label} types={[`${value.type}`]} initialValue={''} onSuggestSelect={() => {}} />
          );
        } else if (value.type === 'fieldArrays') {
          const renderEmployee = ({ fields }) => ( // eslint-disable-line
            <ul className="add-field-list">
              {fields.map((member, index) =>
                <div key={index}>
                  <li>
                    <div className="row">
                      <div className="small-12 medium-6 columns">
                        <input
                          type="text"
                          defaultValue={value.fieldName[i]}
                          onBlur={(event) => this._handleChange(`${member.replace('[', '.').replace(']', '.')}name`, event.target.value)}
                           />
                      </div>
                      <div className="small-12 medium-6 columns">
                        <input
                          type="text"
                          defaultValue={value.fieldPosition[i]}
                          onBlur={(event) => this._handleChange(`${member.replace('[', '.').replace(']', '.')}position`, event.target.value)}
                          />
                      </div>
                    </div>
                    <a
                      className="remove-button"
                      type="button"
                      onClick={() => fields.remove(index)}>Remove
                    </a>
                  </li>
                </div>
                )}
              <div className="add-field">
                <FontAwesome name="plus-circle" />
                <a type="button" onClick={() => fields.push({})}>Add</a>
              </div>
            </ul>
          );
          return (
            <div className="add-input">
              <FieldArray name={value.name} component={renderEmployee} />
            </div>
           );
         }
      })()}
    </div>
  );
};

export {mockChooseEditType};
