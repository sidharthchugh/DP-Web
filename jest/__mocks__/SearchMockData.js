const mockSearchDetails = {
  searchTypes: {
    name: 'searchTypeDropdown',
    label: 'Choose your type of search',
    value: null,
    type: 'dropdown',
    typeValues: {
      auto: 'Automatic - Use my profile',
      productAuto: 'Products/Services - Search interesting options for my company',
      productSpecific: 'Products/Services - I want to specify the details of what I am looking for',
      partnerAuto: 'Strategic partners - Search relevant options for my company',
      partnerSpecific: 'Strategic partners - I want to specify the details of what I am looking for',
      otherAuto: 'Other - I am interested in other relevant opportunities',
      otherSpecific: 'Other - I want to specify other search details'
    },
    id: null
  },
  productSearch: {
    productSearchName: {
      name: 'productSearchName',
      label: 'Search Name',
      value: null,
      type: 'string'
    },
    businessObjectives: {
      name: 'businessObjectives',
      label: 'Business Objectives',
      value: null,
      type: 'multiSelect'
    },
    operationalGoal: {
      name: 'operationalGoal',
      label: 'Operational Goal / Problem',
      value: null,
      type: 'textarea'
    },
    budget: {
      name: 'budget',
      label: 'Budget',
      value: null,
      type: 'string'
    },
    timeline: {
      name: 'timeline',
      label: 'Timeline',
      value: null,
      type: 'string'
    },
    digitalizationScope: {
      name: 'digitalizationScope',
      label: 'Digitalization Scope',
      value: null,
      type: 'textarea'
    },
    capacity: {
      name: 'capacity',
      label: 'CAPACITY REQUIREMENTS',
      value: null,
      type: 'string'
    },
    solution: {
      name: 'solution',
      label: 'Solution',
      value: null,
      type: 'multiSelect'
    },
    valueChain: {
      name: 'valueChain',
      label: 'FOR VALUE CHAIN / SUPPORT FUNCTION',
      value: null,
      type: 'dropdown'
    },
    keyActivities: {
      name: 'keyActivities',
      label: 'PERFORMS KEY ACTIVITIES',
      value: null,
      type: 'multiSelect'
    },
    keyResources: {
      name: 'keyResources',
      label: 'IMPROVES KEY RESOURCES',
      value: null,
      type: 'multiSelect'
    },
    technologiesNeeded: {
      name: 'technologiesNeeded',
      label: 'TECHNOLOGIES NEEDED',
      value: null,
      type: 'multiSelect'
    },
    compatibileTechnologies: {
      name: 'compatibileTechnologies',
      label: 'COMPATIBLE TECHNOLOGIES',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyResources: {
      name: 'partnerKeyResources',
      label: 'Company Description',
      value: null,
      type: 'textarea'
    },
    industryStandards: {
      name: 'industryStandards',
      label: 'PRODUCT / INDUSTRY STANDARDS',
      value: null,
      type: 'multiSelect'
    },
    dataSecurity: {
      name: 'dataSecurity',
      label: 'DATA SECURITY CONCERN',
      value: null,
      type: 'textarea'
    },
    legalConsideration: {
      name: 'legalConsideration',
      label: 'LEGAL CONSIDERATIONS',
      value: null,
      type: 'textarea'
    },
    otherPartnerCharacteristics: {
      name: 'otherPartnerCharacteristics',
      label: 'OTHER PRODUCT REQUIREMENTS',
      value: null,
      type: 'textarea'
    }
  },
  partnerSearch: {
    partnerSearchName: {
      name: 'partnerSearchName',
      label: 'Specific Strategic Partner',
      value: null,
      type: 'string'
    },
    otherPartnerCharacteristics: {
      name: 'otherPartnerCharacteristics',
      label: 'Other Partner Characteristics',
      value: null,
      type: 'multiSelect'
    },
    partnerLanguages: {
      name: 'partnerLanguages',
      label: 'Partner Languages',
      value: null,
      type: 'multiSelect'
    },
    partnerLocation: {
      name: 'partnerLocation',
      label: 'Partner Location',
      value: null,
      type: 'multiSelect'
    },
    partnerSizeIndication: {
      name: 'partnerSizeIndication',
      label: 'Partner Size Indication',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyResources: {
      name: 'partnerKeyResources',
      label: 'Partner Key Resources',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyActivities: {
      name: 'partnerKeyActivities',
      label: 'Partner Key Activites',
      value: null,
      type: 'multiSelect'
    },
    partnerBusinessType: {
      name: 'partnerBusinessType',
      label: 'Partner Business Type',
      value: null,
      type: 'multiSelect'
    },
    partnerSubSector: {
      name: 'partnerSubSector',
      label: 'Partner Sub-Sector',
      value: null,
      type: 'multiSelect'
    },
    partnerSector: {
      name: 'partnerSector',
      label: 'Partner Sector',
      value: null,
      type: 'multiSelect'
    },
    partnerIndustry: {
      name: 'partnerIndustry',
      label: 'Partner Industry',
      value: null,
      type: 'multiSelect'
    },
    partnerBenefits: {
      name: 'partnerBenefits',
      label: 'Benefits to Partner',
      value: null,
      type: 'textarea'
    },
    collaborationRequirements: {
      name: 'collaborationRequirements',
      label: 'Collaboration Requirements',
      value: null,
      type: 'textarea'
    },
    synergies: {
      name: 'synergies',
      label: 'Intended Synergies',
      value: null,
      type: 'textarea'
    },
    assetsSearch: {
      name: 'assetsSearch',
      label: 'Assets we Search',
      value: null,
      type: 'textarea'
    },
    assetsProvide: {
      name: 'assetsProvide',
      label: 'Assets We Provide',
      value: null,
      type: 'textarea'
    },
    operationalGoal: {
      name: 'operationalGoal',
      label: 'Operational Goal / Problem',
      value: null,
      type: 'textarea'
    },
    businessObjectives: {
      name: 'businessObjectives',
      label: 'Business Objectives',
      value: null,
      type: 'multiSelect'
    }
  },
  otherSearch: {
    otherSearchName: {
      name: 'otherSearchName',
      label: 'Other Search',
      value: null,
      type: 'string'
    }
  },
  searchPreferences: {
    suitableProduct: {
      name: null,
      label: 'I am interested in suitable product/service offers',
      value: null,
      type: 'checkbox',
      id: null
    },
    strategicPartners: {
      name: null,
      label: 'I am interested in strategic partnerships',
      value: null,
      type: 'checkbox',
      id: null
    },
    collaboration: {
      name: null,
      label: 'I am interested in joining projects for collaboration',
      value: null,
      type: 'checkbox',
      id: null
    },
    customer: {
      name: null,
      label: 'I am interested in new customers for my product/services',
      value: null,
      type: 'checkbox',
      id: null
    },
    opportunities: {
      name: null,
      label: 'I am interested in other relevant opportunities',
      value: null,
      type: 'checkbox',
      id: null
    }
  }
};

export {
  mockSearchDetails
};
