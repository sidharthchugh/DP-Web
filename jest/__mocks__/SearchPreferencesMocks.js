export const savedPreferences = {
  suitableProduct: false,
  strategicPartners: false,
  collaboration: false,
  customer: false,
  opportunities: false
};
