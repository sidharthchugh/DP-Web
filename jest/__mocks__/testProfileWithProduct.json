{
    "_id" : ObjectId("5853c3eafa9dcae89aa79652"),
    "companyId" : ObjectId("5853c3eafa9dcae89aa79651"),
    "references" : [
        {
            "_id" : ObjectId("5853fb2b45c69401c29aa080"),
            "description" : "description test value",
            "name" : "Project name test value"
        }
    ],
    "projectDescription" : "",
    "projectName" : "",
    "products" : [
        {
            "_id" : ObjectId("5853c3eafa9dcae89aa79653"),
            "companyId" : ObjectId("5853c3eafa9dcae89aa79652"),
            "otherProductAttributes" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea2"),
                    "value" : "other product attributes test value 1",
                    "label" : "other product attributes test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea1"),
                    "value" : "other product attributes test value 2",
                    "label" : "other product attributes test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "segmentsDescription" : "target segments description test value",
            "businessObjective" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea4"),
                    "value" : "solves business objective test value 1",
                    "label" : "solves business objective test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea3"),
                    "value" : "solves business objective test value 2",
                    "label" : "solves business objective test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "dataSecurity" : "data security textarea test value",
            "industryStandards" : "",
            "implementation" : "implementation test value",
            "geographicalMarkets" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea6"),
                    "value" : "geo markets served test value 1",
                    "label" : "geo markets served test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea5"),
                    "value" : "geo markets served test value 2",
                    "label" : "geo markets served test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "maintenanceDifficulty" : "maintenance difficulty & cost test value",
            "integrationDifficulty" : "integration difficulty & related costs test value",
            "delivery" : "delivery/construction time test value",
            "capacityIndication" : "capacity/scalability test value",
            "pricingIndication" : "pricing indication test value",
            "patents" : "patents test value",
            "requirements" : [],
            "compatibilities" : [],
            "technologies" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea8"),
                    "value" : "tech used in the product test value 1",
                    "label" : "tech used in the product test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea7"),
                    "value" : "tech used in the product test value 2",
                    "label" : "tech used in the product test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "uniqueSellingPoints" : "unique selling points test value",
            "valueProposition" : "value proposition test value",
            "productDigitalization" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e33"),
                    "value" : "digitalization scope test value 1",
                    "label" : "digitalization scope test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e32"),
                    "value" : "digitalization scope test value 2",
                    "label" : "digitalization scope test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "keyResources" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e35"),
                    "value" : "reduces/improves key resources test value 1",
                    "label" : "reduces/improves key resources test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e34"),
                    "value" : "reduces/improves key resources test value 2",
                    "label" : "reduces/improves key resources test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "keyActivities" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e37"),
                    "value" : "performs/supports key activities test value 1",
                    "label" : "performs/supports key activities test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e36"),
                    "value" : "performs/supports key activities test value 2",
                    "label" : "performs/supports key activities test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "businessType" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e39"),
                    "value" : "application for business type test value 1",
                    "label" : "application for business type test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e38"),
                    "value" : "application for business type test value 2",
                    "label" : "application for business type test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "applicationValeChain" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3b"),
                    "value" : "value chain position test value 1",
                    "label" : "value chain position test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3a"),
                    "value" : "value chain position test value 2",
                    "label" : "value chain position test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "applicationSector" : "Energy Equipment & Services",
            "applicationIndustry" : "Commercial & Professional Services",
            "problemSolved" : "problem solved test value",
            "solution" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3e"),
                    "value" : "Enterprise Security",
                    "label" : "Enterprise Security",
                    "className" : ""
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3d"),
                    "value" : "Enterprise Networking",
                    "label" : "Enterprise Networking",
                    "className" : ""
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3c"),
                    "value" : "custom production solution value",
                    "label" : "custom production solution value",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "digitalProductCategory" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e40"),
                    "value" : "product category test value 1",
                    "label" : "product category test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3f"),
                    "value" : "product category test value 2",
                    "label" : "product category test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "productName" : "Product name test value"
        }
    ],
    "digitalizationLevelSupport" : "Digital communication & documentation",
    "supportFunction" : "Information Technology",
    "digitalizationLevelSC" : "Digital communication & documentation",
    "valueChain" : "Operations",
    "otherOnline" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e31"),
            "name" : "Other contact value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e30"),
            "name" : "Other contact value 2"
        }
    ],
    "twitter" : "@twitter_testvalue",
    "linkedIn" : "linkedin-testvalue",
    "facebook" : "facebook-testvalue",
    "skype" : "skype.testvalue",
    "email" : "email@testvalue.com",
    "phone" : "123456789",
    "companyWebsite" : "website value",
    "totalFundingObtain" : "999999999",
    "revenue" : "999999999",
    "revenueStreams" : "",
    "channels" : "",
    "customerSegments" : "customer segments text area value",
    "customerRelationships" : "customer relationships text area value",
    "valueProposition" : "value proposition text area value",
    "costStructure" : "Cost structure text area value",
    "keyResources" : "key resources text area value",
    "keyActivities" : "Key activities textarea value",
    "otherBusinessRelationships" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2f"),
            "name" : "Other business relationship name 1",
            "type" : "Other business relationship type 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2e"),
            "name" : "Other business relationship name 2",
            "type" : "Other business relationship type 2"
        }
    ],
    "investors" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2d"),
            "name" : "Investor value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2c"),
            "name" : "Investor value 2"
        }
    ],
    "strategicPartners" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2b"),
            "name" : "Strategic partner name 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2a"),
            "name" : "Strategic partner name 2"
        }
    ],
    "suppliers" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e29"),
            "name" : "Supplier name 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e28"),
            "name" : "Supplier name 2"
        }
    ],
    "customers" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e27"),
            "name" : "Customer name value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e26"),
            "name" : "Customer name value 2"
        }
    ],
    "collaborators" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e25"),
            "name" : "Team member name 1",
            "position" : "Job position 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e24"),
            "name" : "Team member name 2",
            "position" : "Job position 2"
        }
    ],
    "othercompanyAttributes" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e23"),
            "value" : "My other company attribute 1",
            "label" : "My other company attribute 1",
            "className" : "Select-create-option-placeholder"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e22"),
            "value" : "My other company attribute 2",
            "label" : "My other company attribute 2",
            "className" : "Select-create-option-placeholder"
        }
    ],
    "vision" : "My vision value",
    "mission" : "My mission value",
    "technologies" : "",
    "processesOperations" : "",
    "ftes" : "6-10",
    "languages" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e21"),
            "value" : "German",
            "label" : "German",
            "className" : ""
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e20"),
            "value" : "Hebrew",
            "label" : "Hebrew",
            "className" : ""
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1f"),
            "value" : "Hindi",
            "label" : "Hindi",
            "className" : ""
        }
    ],
    "streetAddress" : "Ackerstraße",
    "zipCode" : "10115",
    "homeMarket" : "Europe",
    "locations" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1e"),
            "value" : "Location value 1",
            "label" : "Location value 1",
            "className" : "Select-create-option-placeholder"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1d"),
            "value" : "Location value 2",
            "label" : "Location value 2",
            "className" : "Select-create-option-placeholder"
        }
    ],
    "country" : "Germany",
    "headquarters" : "Berlin",
    "yearEstablished" : "2016",
    "subSector" : "Personal Products",
    "sector" : "IT Services",
    "industry" : "Information Technology",
    "legalForm" : "Limited partnership",
    "businessType" : "Software producer",
    "organizationType" : "Small Enterprise",
    "companyDescription" : "My test company description value",
    "slogan" : "My test slogan value",
    "statusLive": "Not Live",
    "companyName" : "Company Name test",
    "logoURI" : "companylogo-5853c3eafa9dcae89aa79652.png",
    "__v" : 0
}
