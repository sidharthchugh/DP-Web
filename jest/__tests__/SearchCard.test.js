import {mockSearchDetails} from '../__mocks__/SearchMockData';
import {SearchCard} from '../../src/components/Search/SearchCard';

const searchCard = new SearchCard();

describe('SearchCard', () => {
  it('merges Search field types to the expected structure', () => {
    expect(searchCard.mergeSearchFields()).toMatchSnapshot();
  });
});
