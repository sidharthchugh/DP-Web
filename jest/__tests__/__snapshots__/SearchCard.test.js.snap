exports[`SearchCard merges Search field types to the expected structure 1`] = `
Object {
  "otherDetailSearch": Object {
    "fields": Object {
      "operationalGoal": Object {
        "label": "Operational Goal / Problem",
        "name": "operationalGoal",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "otherDetailSearchName": Object {
        "label": "Other Search",
        "name": "otherDetailSearchName",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "searchDescription": Object {
        "label": "Other Search Description",
        "name": "searchDescription",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "searchObjective": Object {
        "label": "Search Objective",
        "name": "searchObjective",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "searchRelationship": Object {
        "label": "Search Relationship Type",
        "name": "searchRelationship",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
    },
    "meta": Object {
      "createdAt": null,
      "name": "otherDetailSearch",
      "type": "Other - I want to specify other search details",
    },
  },
  "otherSearch": Object {
    "fields": Object {},
    "meta": Object {
      "createdAt": null,
      "name": "otherSearch",
      "type": "Other - I am interested in other relevant opportunities",
    },
  },
  "partnerDetailSearch": Object {
    "fields": Object {
      "assetsProvide": Object {
        "label": "Assets We Provide",
        "name": "assetsProvide",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "assetsSearch": Object {
        "label": "Assets we Search",
        "name": "assetsSearch",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "businessObjectives": Object {
        "label": "Business Objectives",
        "name": "businessObjectives",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "collaborationRequirements": Object {
        "label": "Collaboration Requirements",
        "name": "collaborationRequirements",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "operationalGoal": Object {
        "label": "Operational Goal / Problem",
        "name": "operationalGoal",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "otherPartnerCharacteristics": Object {
        "label": "Other Partner Characteristics",
        "name": "otherPartnerCharacteristics",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerBenefits": Object {
        "label": "Benefits to Partner",
        "name": "partnerBenefits",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "partnerBusinessType": Object {
        "label": "Partner Business Type",
        "name": "partnerBusinessType",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerDetailSearchName": Object {
        "label": "Specific Strategic Partner",
        "name": "partnerDetailSearchName",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "partnerIndustry": Object {
        "label": "Partner Industry",
        "name": "partnerIndustry",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerKeyActivities": Object {
        "label": "Partner Key Activites",
        "name": "partnerKeyActivities",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerKeyResources": Object {
        "label": "Partner Key Resources",
        "name": "partnerKeyResources",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerLanguages": Object {
        "label": "Partner Languages",
        "name": "partnerLanguages",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerLocation": Object {
        "label": "Partner Location",
        "name": "partnerLocation",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerSector": Object {
        "label": "Partner Sector",
        "name": "partnerSector",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerSizeIndication": Object {
        "label": "Partner Size Indication",
        "name": "partnerSizeIndication",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "partnerSubSector": Object {
        "label": "Partner Sub-Sector",
        "name": "partnerSubSector",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "synergies": Object {
        "label": "Intended Synergies",
        "name": "synergies",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
    },
    "meta": Object {
      "createdAt": null,
      "name": "partnerDetailSearch",
      "type": "Strategic partners - I want to specify the details of what I am looking for",
    },
  },
  "partnerSearch": Object {
    "fields": Object {},
    "meta": Object {
      "createdAt": null,
      "name": "partnerSearch",
      "type": "Strategic partners - Search relevant options for my company",
    },
  },
  "productDetailSearch": Object {
    "fields": Object {
      "budget": Object {
        "label": "Budget",
        "name": "budget",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "businessObjectives": Object {
        "label": "Business Objectives",
        "name": "businessObjectives",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "capacity": Object {
        "label": "Capacity Requirements",
        "name": "capacity",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "companyDescription": Object {
        "label": "Company Description",
        "name": "companyDescription",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "compatibileTechnologies": Object {
        "label": "Compatibile Technologies",
        "name": "compatibileTechnologies",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "dataSecurity": Object {
        "label": "Data Security Concern",
        "name": "dataSecurity",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "digitalizationScope": Object {
        "label": "Digitalization Scope",
        "name": "digitalizationScope",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "industryStandards": Object {
        "label": "Product / Industry Standards",
        "name": "industryStandards",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "keyActivities": Object {
        "label": "Performs Key Activities",
        "name": "keyActivities",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "keyResources": Object {
        "label": "Improves Key Resources",
        "name": "keyResources",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "legalConsideration": Object {
        "label": "Legal Considerations",
        "name": "legalConsideration",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "operationalGoal": Object {
        "label": "Operational Goal / Problem",
        "name": "operationalGoal",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "otherProductRequirements": Object {
        "label": "Other Product Requirements",
        "name": "otherProductRequirements",
        "tooltip": "",
        "type": "textarea",
        "value": null,
      },
      "productCategory": Object {
        "label": "Product Category",
        "name": "productCategory",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "productDetailSearchName": Object {
        "label": "Search Name",
        "name": "productDetailSearchName",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "solution": Object {
        "label": "Solution",
        "name": "solution",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "technologiesNeeded": Object {
        "label": "Technologies Needed",
        "name": "technologiesNeeded",
        "tooltip": "",
        "type": "multiSelect",
        "typeValues": Array [
          Object {},
        ],
        "value": null,
      },
      "timeline": Object {
        "label": "Timeline",
        "name": "timeline",
        "tooltip": "",
        "type": "string",
        "value": null,
      },
      "valueChain": Object {
        "label": "For Value Chain / Support Function",
        "name": "valueChain",
        "tooltip": "",
        "type": "dropdown",
        "typeValues": Array [],
        "value": null,
      },
    },
    "meta": Object {
      "createdAt": null,
      "name": "productDetailSearch",
      "type": "Products/Services - I want to specify the details of what I am looking for",
    },
  },
  "productSearch": Object {
    "fields": Object {},
    "meta": Object {
      "createdAt": null,
      "name": "productSearch",
      "type": "Products/Services - Search interesting options for my company",
    },
  },
  "searchPreferences": Object {
    "collaboration": Object {
      "label": "I am interested in joining projects for collaboration",
      "name": "collaboration",
      "type": "checkbox",
      "value": null,
    },
    "customer": Object {
      "label": "I am interested in new customers for my product/services",
      "name": "customer",
      "type": "checkbox",
      "value": null,
    },
    "opportunities": Object {
      "label": "I am interested in other relevant opportunities",
      "name": "opportunities",
      "type": "checkbox",
      "value": null,
    },
    "strategicPartners": Object {
      "label": "I am interested in strategic partnerships",
      "name": "strategicPartners",
      "type": "checkbox",
      "value": null,
    },
    "suitableProduct": Object {
      "label": "I am interested in suitable product/service offers",
      "name": "suitableProduct",
      "type": "checkbox",
      "value": null,
    },
  },
  "searchTypes": Object {
    "label": "CHOOSE A SEARCH SCENARIO",
    "name": "searchTypeDropdown",
    "type": "dropdown",
    "typeValues": Object {
      "other": "Other - I am interested in other relevant opportunities",
      "otherDetail": "Other - I want to specify other search details",
      "partner": "Strategic partners - Search relevant options for my company",
      "partnerDetail": "Strategic partners - I want to specify the details of what I am looking for",
      "product": "Products/Services - Search interesting options for my company",
      "productDetail": "Products/Services - I want to specify the details of what I am looking for",
    },
    "value": null,
  },
}
`;
